.circleci/config.yml

```diff
@@ -0,0 +1,124 @@
+          requires:
+            - code_style
+            - prettier
+            - build_and_test
+          filters:
+            branches:
+              only:
+                - master
+
+jobs:
+  code_style:
+    docker:
+      - image: << pipeline.parameters.ubuntu_docker >>
+    steps:
+      - checkout
+      - install_dependencies
+      - run: npm run lint
+
+  prettier:
+    docker:
+      - image: << pipeline.parameters.ubuntu_docker >>
+    steps:
+      - checkout
+      - install_dependencies
+      - run: |
+          npm install prettier
+          npx prettier --check .
+
+  build_and_test:
+    parameters:
+      browser:
+        type: string
+    machine:
+      image: << pipeline.parameters.ubuntu_machine >>
+    resource_class: large
+    steps:
+      - checkout
+      - install_dependencies:
+          sudo: 'sudo'
+      - run:
+          name: Install docker
+          command: |
+            export SUDO=sudo
+            $SUDO apt update
+            $SUDO apt -y install ca-certificates curl gnupg lsb-release
+            $SUDO curl -fsSL https://download.docker.com/linux/ubuntu/gpg | $SUDO gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+            $SUDO echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
+              $(lsb_release -cs) stable" | $SUDO tee /etc/apt/sources.list.d/docker.list > /dev/null
+            $SUDO apt update
+            $SUDO apt -y install docker-ce docker-ce-cli containerd.io
+      - run:
+          name: Compile test
+          command: |
+            npm run build
+            npx tsc -p tests/compile
+      - browser-tools/install-browser-tools
+      - run:
+          name: << parameters.browser >> test
+          command: |
+            if [ "<< parameters.browser >>" == "node" ]; then unset TEST_BROWSER; else export TEST_BROWSER=<< parameters.browser >> ; fi
+            npm test
+            make docker-test
+
+  generate_docs:
+    docker:
+      - image: << pipeline.parameters.ubuntu_docker >>
+    steps:
+      - checkout
+      - install_dependencies
+      - run:
+          name: GitHub Pages
+          command: |
+            touch docs/.nojekyll
+            npm run docs
+            npm install -g --silent gh-pages@2.0.1
+            # git config user.email "ci-build@algorand.com"
+            # git config user.name "ci-build"
+
+commands:
+  install_dependencies:
+    parameters:
+      sudo:
+        type: string
+        default: ''
+    steps:
+      - run:
+          name: Install Dependencies
+          command: |
+            << parameters.sudo >> apt -y update && << parameters.sudo >> apt -y upgrade
+            << parameters.sudo >> apt -y install curl make git build-essential jq unzip
+      - node/install:
+          node-version: '12'
+      - run:
+          name: npm install
+          command: |
+            set -e
+            npm install
```

Comment:
[`npm ci`](https://docs.npmjs.com/cli/v8/commands/npm-ci) should be used instead. Otherwise a different version of our dependencies might get installed

Reply:
updated ty